### ISSUE TITLE
Integrate Z-Frame registration into SliceTracker

### ISSUE THREAD

**Issue by che85** [SlicerProstate/SliceTracker#86]
In order do completely replace the currently used approach, we need to add the Z-Frame registration as follows:

1. As soon as the template image arrives in SliceTracker, Z-Frame registration should automatically be executed
2. After Cover Prostate registration, display Z-Frame model with appropriate coordinates from NeedleGuidedTemplate (https://github.com/tokjun/NeedleGuideTemplate/tree/master/NeedleGuideTemplate)

**Comment by che85** [comment]
@fedorov Did I forget anything? 

Maybe we also should add a checkbox for displaying/hiding "Z-Frame model"?

**Comment by fedorov** [comment]
Yes:

* once z-frame registration is completed, should show z-frame and display of the z-frame image, with the z-frame model intersection with the slice to confirm registration accuracy, before the operator accepts the result
* on arrival of the "Cover prostate" image, show model of the template and 3-view to check prostate coverage
* UI should have check-boxes to enable visualization of the needle (for the selected target) and template; for needle visualization, slice intersection with the images should be enabled to allow evaluation of the deviation of the actual needle path from the planned path

Basically, all those features we utilize during the cases.

@tokjun do you have any features on your wish-list that I missed or that we should consider?

**Comment by che85** [comment]
@fedorov I think we have two options to display information about which hole to use for "selected" targets. a) I would add a table with headers (Hole, Depth, Position(RAS) ) which would be displayed for the selected fiducial. b) Add "Hole" and "Depth" to the current fiducial table which would immediately be updated after registration. 

What do you think?

**Comment by fedorov** [comment]
When linemarkerregistration is not available, the module should fail gracefully. Right now, it is just stuck with the progress bar, since there is unhandled error due to the missing module

```
Traceback (most recent call last):
  File "/Users/fedorov/github/SliceTracker/SliceTracker.py", line 2197, in importDICOMSeries
    self._incomingDataCallback(newList=newFileList)
  File "/Users/fedorov/github/SliceTracker/SliceTracker.py", line 1735, in onNewImageDataReceived
    self.onTrackTargetsButtonClicked()
  File "/Users/fedorov/github/SliceTracker/SliceTracker.py", line 1560, in onTrackTargetsButtonClicked
    self.openZFrameRegistrationStep()
  File "/Users/fedorov/github/SliceTracker/SliceTracker.py", line 1579, in openZFrameRegistrationStep
    self.logic.runZFrameRegistration(volume)
  File "/Users/fedorov/github/SliceTracker/SliceTracker.py", line 2366, in runZFrameRegistration
    registration.runRegistration()
  File "/Users/fedorov/github/SliceTracker/SliceTrackerUtils/ZFrameRegistration.py", line 38, in runRegistration
    slicer.cli.run(slicer.modules.linemarkerregistration, None, params, wait_for_completion=True)
AttributeError: 'module' object has no attribute 'linemarkerregistration'
```
